ngx_http_pipelog_module.c

```diff
@@ -2238,15 +2266,18 @@ static ngx_int_t
 init_module (ngx_cycle_t *cycle) {
     ngx_http_pipelog_main_conf_t *pmcf;
     ngx_fd_t fd;
-
     if (!ngx_test_config) {
         pmcf = ngx_http_cycle_get_module_main_conf(cycle, ngx_http_pipelog_module);
-        pmcf->pid = fork();
-        if (pmcf->pid < 0) {
-            ngx_log_error(NGX_LOG_ALERT, cycle->log, 0, "%s: fork(): error", MODULE_NAME);
+        switch (pmcf->pid = fork())
+        {
+        case -1:
+            ngx_log_error(NGX_LOG_ALERT, cycle->log, 0, "%s: init_module fork(): error", MODULE_NAME);
             return NGX_ERROR;
-        }
-        if (pmcf->pid == 0) {
+        case 0:
+            if(setsid() == -1) {
```

Comment:
I think it might be enough to create a process group without creating a session.

Reply:
setsid() was just handy to do, so it puts them all in the same PGID.